### PR TITLE
Spinner buttons

### PR DESCRIPTION
**Pull request by mosch**
This PR allows the floating button to indicate that something is in progress

screenshot 2018-10-04 at 09 47 34

Checklist:
- [x] Test added / Snapshots updated
- [x] Documentation added / updated

### CODE REVIEW

File: src/molecules/SquareIconButton.jsx [hunk 1]
```diff
@@ -63,8 +62,4 @@ class SquareIconButton extends React.Component {
   }
 }
 
-const mapThemeToProps = theme => ({
-  textColor: theme.text,
-})
-
-export default withTheme(mapThemeToProps)(SquareIconButton)
+export default SquareIconButton
```
Comment (fivenp) [inline src/molecules/SquareIconButton.jsx]: are you sure this doesn't do any harm?!
stuff like - "oh.. now the icon - color is dark on our AppMenu"

Comment (mosch) [reply src/molecules/SquareIconButton.jsx]: Imho it wasn't used at all – so I think this doesn't change anything?

Comment (fivenp) [reply src/molecules/SquareIconButton.jsx]: lets remove it and simply give it a try before upgrading :)

**Comment by dan-lee** [comment]
Should we disable buttons which are marked in progress implicitly? I can't imagine a a use case where it should be in progress and still be clickable.

**Comment by dan-lee** [comment]
Also Storybook → Button → in progress doesn't really show anything.

File: src/molecules/FloatingButton.jsx [hunk 2]
```diff
@@ -36,13 +60,20 @@ class FloatingButton extends React.Component {
...
             height: 50,
             overflow: 'hidden',
             width: '100%',
           })}
         >
-          <button {...buttonStyle} {...props} />
+          <button
+            {...buttonStyle}
+            {...props}
+            onClick={inProgress || onClick}
```
Comment (dan-lee) [inline src/molecules/FloatingButton.jsx]: ?

File: src/molecules/FloatingButton.jsx [hunk 3]
```diff
@@ -13,17 +14,40 @@ const buttonStyle = css({
...
+    const {
+      color,
+      onClick,
+      disabled,
+      disabledColor,
+      inProgress,
+      children,
+      ...props
+    } = this.props
+    const isDisabled = disabled || inProgress
```
Comment (dan-lee) [inline src/molecules/FloatingButton.jsx]: Please add a comment for why we need this.

**Pull request merged**